crates/wasmparser/src/validator/operators.rs

```diff
@@ -90,29 +90,53 @@ struct Locals {
     all: Vec<(u32, ValType)>,
 }
 
+/// A Wasm control flow block on the control flow stack during Wasm validation.
+//
+// # Dev. Note
+//
 // This structure corresponds to `ctrl_frame` as specified at in the validation
 // appendix of the wasm spec
 #[derive(Debug)]
-struct Frame {
-    // Indicator for what kind of instruction pushed this frame.
-    kind: FrameKind,
-    // The type signature of this frame, represented as a singular return type
-    // or a type index pointing into the module's types.
-    block_type: BlockType,
-    // The index, below which, this frame cannot modify the operand stack.
-    height: usize,
-    // Whether this frame is unreachable so far.
-    unreachable: bool,
+pub struct Frame {
+    /// Indicator for what kind of instruction pushed this frame.
+    pub kind: FrameKind,
```

Comment:
I was not sure if we want to have public fields or getter methods instead. The style of the codebase uses public fields mostly so that's what I did.

Reply:
Yeah I think this is fine